services/src/main/java/org/keycloak/services/resources/admin/ClientResource.java

```diff
@@ -514,8 +514,10 @@ public Stream<UserSessionRepresentation> getUserSessions(@QueryParam("first") In
 
         firstResult = firstResult != null ? firstResult : -1;
         maxResults = maxResults != null ? maxResults : Constants.DEFAULT_MAX_RESULTS;
-        return session.sessions().getUserSessionsStream(client.getRealm(), client, firstResult, maxResults)
-                .map(ModelToRepresentation::toRepresentation);
+        return session.sessions()
+                .getUserSessionsStream(client.getRealm(), client, firstResult, maxResults)
+                .map(ModelToRepresentation::toRepresentation)
+                .filter(Objects::nonNull);
```

Comment:
It should be safer to filter sessions in `InfinispanUserSessionProvider` and avoid changing its callers.

Reply:
I am not sure about this.
The consequence of your suggestion is to tightly couple the `UserSessionProvider` and the `UserProvider`.
It must be defined in the interface contract what happens to sessions without a valid user (what about deactivated users?). But then the function should be renamed to something like "_getValidUserSessionsStream()_".
Before changing the interface in such a way, all users of this interface must be looked at.

Reply:
Looking at the places where the method is used, they only want valid sessions.

If we need to for whatever reason to handle the cases where sessions are not bound to users, we can provide a separate method or specific handing within the session provider itself.